### PR TITLE
activate batch pre-emption

### PR DESCRIPTION
**Pull request by mj3ai**

### CODE REVIEW

File: nomad/nomad_node_configs/initial_server.hcl [hunk 1]
```diff
@@ -4,6 +4,12 @@ bind_addr = "0.0.0.0"
 server {
   enabled = true
   bootstrap_expect = 1
+  preemption_config {
+    batch_scheduler_enabled    = true
+    system_scheduler_enabled   = true
+    service_scheduler_enabled  = false
```
Comment (ghost) [inline nomad/nomad_node_configs/initial_server.hcl]: Having this set to false will not preempt deployment jobs. Currently higher priority train jobs will preempt only lower priority train jobs, but not deployments. Which is fine for this PR if that's the intended behavior

File: nomad/nomad_node_configs/initial_server.hcl [hunk 2]
```diff
@@ -4,6 +4,12 @@ bind_addr = "0.0.0.0"
 server {
   enabled = true
   bootstrap_expect = 1
+  preemption_config {
```
Comment (ghost) [inline nomad/nomad_node_configs/initial_server.hcl]: https://developer.hashicorp.com/nomad/docs/configuration/server#bootstrapping-with-a-custom-scheduler-config

Does the preemption_config need to be in the default_scheduler_config stanza, or does this work?

File: nomad/nomad_node_configs/initial_server.hcl [hunk 3]
```diff
@@ -4,6 +4,12 @@ bind_addr = "0.0.0.0"
...
+    batch_scheduler_enabled    = true
+    system_scheduler_enabled   = true
```
Comment (ghost) [inline nomad/nomad_node_configs/initial_server.hcl]: Why is this true? We shouldn't need to preempt system jobs right?

File: nomad/nomad_node_configs/initial_server.hcl (hunk 1 above)
Comment (mj3ai) [reply nomad/nomad_node_configs/initial_server.hcl]: Yes... that was the intended behavior.

File: nomad/nomad_node_configs/initial_server.hcl (hunk 2 above)
Comment (mj3ai) [reply nomad/nomad_node_configs/initial_server.hcl]: You are right... I had missed that. Corrected now.

File: nomad/nomad_node_configs/initial_server.hcl (hunk 3 above)
Comment (mj3ai) [reply nomad/nomad_node_configs/initial_server.hcl]: Preemption of system jobs is enabled by default. Hence, I left it as such. Should we disable it?

Comment (ghost) [reply nomad/nomad_node_configs/initial_server.hcl]: Ok, if it's on by default that's fine. We also don't have any system jobs, so this won't affect us either way.

File: nomad/nomad_node_configs/initial_server.hcl (hunk 1 above)
Comment (ghost) [reply nomad/nomad_node_configs/initial_server.hcl]: Ok, sounds good

**Review by ghost — APPROVED** [review]

**Pull request merged**